src/mpi-base.jl

```diff
@@ -156,6 +120,54 @@ function Comm_size(comm::Comm)
     Int(size[])
 end
 
+function Type_create_struct{T <: Any}(::Type{T})  # <: Any effectively 
```

Comment:
Since `T` is dynamic (i.e. the compiler won't be able to optimize anythin), you could also simply write
```Julia
function Type_create_struct(T::Type)
```
instead